diff --git a/rabit/src/allreduce_robust.cc b/rabit/src/allreduce_robust.cc
index ce5a561..926bc6c 100644
--- a/rabit/src/allreduce_robust.cc
+++ b/rabit/src/allreduce_robust.cc
@@ -52,7 +52,9 @@ void AllreduceRobust::Shutdown(void) {
                 "Shutdown: check ack must return true");
 
 #if defined (__APPLE__)
-        sleep(1);
+  if (world_size > 1) {
+      sleep(1);
+  }
 #endif
 
   AllreduceBase::Shutdown();
