diff --git a/rabit/src/allreduce_robust.cc b/rabit/src/allreduce_robust.cc
index 8ae1acc..20ceafd 100644
--- a/rabit/src/allreduce_robust.cc
+++ b/rabit/src/allreduce_robust.cc
@@ -71,7 +71,9 @@ bool AllreduceRobust::Shutdown(void) {
       ActionSummary::kSpecialOp, cur_cache_seq), "Shutdown: check ack must return true");
 // travis ci only osx test hang
 #if defined (__APPLE__)
-    sleep(1);
+    if (world_size > 1) {
+      sleep(1);
+    }
 #endif
     shutdown_timeout = true;
     if (rabit_timeout_task.valid()) {
